### PR TITLE
No main entry

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

Lib structure changes:
 - Expose hslayers lib functionality via `secondary entry points only`. Main entry point is empty
 - Flatten folder structure by moving `assets`, `css `and `test files` outside of the `src `folder
 - `src` folder was replaced by `core` which includes former layout, map components and core service. 
 
Core basically replaced hslayers main entry point while allowing to simplify directory

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: projects/hslayers/test/hslayers.component.spec.ts [hunk 1]
```diff
@@ -6,7 +6,10 @@ import {CustomTranslationService} from 'hslayers-ng/services/language';
 import {HsConfig} from 'hslayers-ng/config';
 import {HsConfigMock} from './config.service.mock';
 import {HsLayerUtilsService} from 'hslayers-ng/services/utils';
-import {HsMapHostDirective, HslayersComponent} from 'hslayers-ng';
+import {
+  HsMapHostDirective,
+  HslayersComponent,
+} from 'hslayers-ng/core/public-api';
```
Comment (jmacura) [inline projects/hslayers/test/hslayers.component.spec.ts:12]:
```suggestion
} from 'hslayers-ng/core';
```

File: projects/hslayers/components/sidebar/sidebar.component.ts [hunk 2]
```diff
@@ -133,7 +131,7 @@ export class HsSidebarComponent implements OnInit, OnDestroy {
       !this.HsLayoutService.sidebarExpanded;
     this.HsLayoutService.updPanelSpaceWidth();
     setTimeout(() => {
-      this.HsCoreService.updateMapSize();
+      this.HsEventBusService.updateMapSize.next();
```
Comment (jmacura) [inline projects/hslayers/components/sidebar/sidebar.component.ts:134]: I know it was not changed in this PR, but I propose to follow a common convention and rename `updateMapSize` to `mapSizeUpdates`. What do you think?

**Review by jmacura — APPROVED** [review]
Thanks! Overall looks good to me.
One idea crossing my mind is why we don't put config.service also into /core? Does it make sense to import it separately? Or vice versa, does it make sense to use HSLayers without config..?

**Comment by FilipLeitner** [comment]
> Thanks! Overall looks good to me. One idea crossing my mind is why we don't put config.service also into /core? Does it make sense to import it separately? Or vice versa, does it make sense to use HSLayers without config..?

Config is being imported all around the lib and with it being a part of core we would easily end up having some sort of circular dependency which is not even allowed on entry point level eg.  core - map(service) - core.  Basically it would not be possible to import config in any of the entrypoints which are used in core (layout, utils, language, share etc)

**Comment by jmacura** [comment]
> 
> Config is being imported all around the lib and with it being a part of core we would easily end up having some sort of circular dependency as is not even allowed on entry point level eg. core - map(service) - core. Basically it would not be possible to import config in any of the entrypoints which are used in core (layout, utils, language, share etc)

Get that, thanks for clarification!

**Pull request merged**